### DISCUSSION TITLE
subcommands documentation

### DISCUSSION THREAD

**Discussion by aborruso** [nushell/nushell#3474] [Q&A] [answered]
Hi,
where can I find the documentation for the subcommands. 

Thank you

**Comment by sophiajt** [comment]
https://www.nushell.sh/book/command_reference.html

**Reply by aborruso** [reply]
@jonathandturner this is the command documentation page. I am looking for the one for sub-commands.

If there is not a webpage, how to have it via shell?

Thank you

**Reply by sophiajt** [reply]
The subcommands should be in that list as well, though it seems to depend a lot on how the base command is written. If you look for the `to` command it links to the subcommands, and you can find subcommands like `to json`.

In the shell, you can do `help commands | get subcommands`

**Comment by aborruso** [comment, marked as answer]
The way I prefer is this one, `command subcommand --help`

In example `str find-replace --help` gives to you

```
finds and replaces text

Usage:
  > str find-replace <find> <replace> ...args {flags}

Parameters:
  <find> the pattern to find
  <replace> the replacement pattern
  ...args: optionally find and replace text by column paths

Flags:
  -h, --help: Display this help message
  -a, --all: replace all occurrences of find string

Examples:
  Find and replace contents with capture group
  > echo 'my_library.rb' | str find-replace '(.+).rb' '$1.nu'

  Find and replace all occurrences of find string
  > echo 'abc abc abc' | str find-replace -a 'b' 'z'
```

Thank you to the @jonathandturner [reply](https://github.com/nushell/nushell/discussions/3475#discussioncomment-773219).

**Comment by aborruso** [comment]
And now we have it in the help https://www.nushell.sh/cookbook/help.html#specific-information-on-a-command-subcommand
